src/components/rtoken-setup/basket/PrimaryBasket.tsx

```diff
@@ -63,19 +69,34 @@ const PrimaryBasket = ({
   const basket = useAtomValue(basketAtom)
   const units = Object.keys(basket)
   const collateralYield = useAtomValue(collateralYieldAtom)
+  const usdPrice = 1
+  const ethPrice = useAtomValue(ethPriceAtom)
+  const btcPrice = useAtomValue(btcPriceAtom)
+  const eurPrice = useAtomValue(eurPriceAtom)
+
+  let primeBasketUsd = 0
 
   const getEstApy = useCallback(() => {
-    return Object.keys(basket).reduce((prev, current) => {
-      const currentBasket = basket[current]
+    return (
+      Object.keys(basket).reduce((prev, current) => {
+        const currentUnitInUsd = eval(`${current.toLowerCase()}Price`)
+        const currentBasket = basket[current]
+
+        const unitValueInBasket = currentUnitInUsd * +currentBasket.scale
+        primeBasketUsd += unitValueInBasket
 
-      for (let i = 0; i < currentBasket.collaterals.length; i++) {
-        prev +=
-          (collateralYield[currentBasket.collaterals[i].symbol.toLowerCase()] ||
-            0) * (+currentBasket.distribution[i] / 100 || 0)
-      }
+        for (let i = 0; i < currentBasket.collaterals.length; i++) {
+          prev +=
+            (collateralYield[
+              currentBasket.collaterals[i].symbol.toLowerCase()
+            ] || 0) *
+            unitValueInBasket *
+            (+currentBasket.distribution[i] / 100 || 0)
+        }
 
-      return prev
-    }, 0)
+        return prev
+      }, 0) / primeBasketUsd
```

Comment:
I think is safer if we add a default value, like primeBasketUsd || 1

Because its initialized with 0, there may be an edge case where we have a division by 0 crash